src/core/dirlistjob.cpp

```diff
@@ -62,6 +63,12 @@ void DirListJob::exec() {
             false
     };
     if(enu) {
+        bool bindCutFiles = false;
+        if(cutFilesHashSet_
+                && cutFilesHashSet_->size() < 450 /* give up if more than 450 cut files */) {
```

Comment:
> but be aware it's very different code path when selecting & cutting files inside a directory and go to a directory with already selected & cut files

I know but don't see any extra CPU usage in the latter case with ~4000 cut files. Do you see any?